servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/TimeoutHttpRequesterFilter.java

```diff
@@ -100,4 +167,59 @@ public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strat
         // No influence since we do not block.
         return strategy;
     }
+
+    @FunctionalInterface
+    public interface TimeoutFromRequest extends Function<HttpRequestMetaData, Duration> {
```

Comment:
do we need this additional interface? IIUC I wonder if it is just for `@Nullable` annotations, and we have deferred adding new types for this in the past. I we think this provides enough value should we add a more general API (e.g. `NullReturnFunction`) so we don't have to create new types for each use case?

Reply:
The `@Nullable` and documentation are the primary value. Removing the `@Nullable` means that static analysis follows the potentially null reference and may issue warnings if you try to use the result without checking for null. The use in the `TimeoutHttpRequesterFilter` and `TimeoutHttpServiceFilter` both have explicit null checks so are fine. I wouldn't expect this type to be used many other places, if at all.

Reply:
The interface now also extends `HttpExecutionStrategyInfluencer` and for documentation, discoverability reasons  and (to be honest) so that it can be implemented by an anonymous inner class I chose to keep it as an explicit interface.